main_test.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInitLogger(t *testing.T) {
+	err := initLogger("debug")
+	assert.Nil(t, err, "failed to initLogger: %s", err)
+}
+
+func TestMain(t *testing.T) {
+	cmd := exec.Command(os.Args[0], "--proxy-winsize")
```

Comment:
Another approach would be to change the shim (in fact maybe all the kata components) to use https://github.com/urfave/cli because you can then create tests by manipulating the `cli.Context`'s like the following which tests `cc-runtime state`:

- https://github.com/clearcontainers/runtime/blob/master/state_test.go#L28..L47

See also:

- https://github.com/clearcontainers/runtime/blob/master/main.go#L337